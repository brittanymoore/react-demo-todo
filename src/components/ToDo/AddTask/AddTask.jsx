import React from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';

export class AddTask extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			name: ''
		};
		this.API_URL = process.env.API_URL;

		this.handleInputChange = this.handleInputChange.bind(this);
		this.handleFormSubmit = this.handleFormSubmit.bind(this);
	}

	handleInputChange(event) {
		const target = event.target;
		const value = target.value;
		const name = target.name;
		this.setState({
			[name]: value
		});
	}

	handleFormSubmit(event) {
		event.preventDefault();
		axios.post(`${this.API_URL}/tasks`, this.state)
			.then((res) => {
				this.props.onAdd(res.data);
				this.setState({ name: '' });
			});
	}

	render() {

		const { name } = this.state;
		const canSubmit = name && name.length > 0;

		return (
			<form noValidate onSubmit={this.handleFormSubmit}>
				<label htmlFor="taskName">
					Add a task:
          <input type="text" name="name" id="taskName" value={this.state.name} onChange={this.handleInputChange} />
				</label>
				<button type="submit" disabled={!canSubmit}>Add</button>
			</form>
		);
	}
}

AddTask.propTypes = {
	onAdd: PropTypes.func
}
