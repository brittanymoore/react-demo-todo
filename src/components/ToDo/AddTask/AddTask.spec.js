import React from 'react'
import { AddTask } from './AddTask'
import renderer from 'react-test-renderer'

test('Create component', () => {
  const onAdd = () => {}
  const component = renderer.create(<AddTask onAdd={onAdd} />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
