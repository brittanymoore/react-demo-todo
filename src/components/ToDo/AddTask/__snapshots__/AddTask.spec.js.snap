// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create component 1`] = `
<form
  noValidate={true}
  onSubmit={[Function]}
>
  <label>
    Add a task:
    <input
      id="taskName"
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <button
    disabled={true}
    type="submit"
  >
    Add
  </button>
</form>
`;
